transport/http.go

```diff
@@ -187,12 +231,12 @@ func NewHTTPTransport() (*HTTPTransport, error) {
 		intakeHeaders:  intakeHeaders,
 		profileHeaders: profileHeaders,
 	}
-	if apiKey := os.Getenv(envAPIKey); apiKey != "" {
-		t.SetAPIKey(apiKey)
-	} else if secretToken := os.Getenv(envSecretToken); secretToken != "" {
-		t.SetSecretToken(secretToken)
+	if opts.APIKey != "" {
+		t.SetAPIKey(opts.APIKey)
+	} else if opts.SecretToken != "" {
+		t.SetSecretToken(opts.SecretToken)
 	}
-	t.SetServerURL(serverURLs...)
+	t.SetServerURL(opts.ServerURLs...)
```

Comment:
IIANM this will panic if HTTPTransportOptions.ServerURLs is empty. This should probably also return an error.

Maybe worth adding a Validate method, like:

```go
func (o HTTPTransportOptions) Validate() error {
    if len(o.ServerURLs) == 0 {
        // ...
    }
    if len(o.ServerTimeout) < 0 {
        // ...
    }
    return nil
}
```